protocol/communities_messenger_signers_test.go

```diff
@@ -824,3 +844,156 @@ func (s *MessengerCommunitiesSignersSuite) TestWithoutMintedOwnerTokenMakingDevi
+	}, nil)
+	s.Require().NoError(err)
+
+	err = s.john.AddCommunityToken(community.IDString(), int(testChainID1), tokenMasterTokenAddress)
+	s.Require().NoError(err)
+
+	// set john as contract owner
+	s.collectiblesServiceMock.SetSignerPubkeyForCommunity(community.ID(), common.PubkeyToHex(&s.john.identity.PublicKey))
+	s.collectiblesServiceMock.SetMockCollectibleContractData(testChainID1, ownerTokenAddress,
+		&communities.CollectibleContractData{TotalSupply: &bigint.BigInt{}})
+	s.collectiblesServiceMock.SetMockCollectibleContractData(testChainID1, tokenMasterTokenAddress,
+		&communities.CollectibleContractData{TotalSupply: &bigint.BigInt{}})
+
+	community, err = s.john.communitiesManager.GetByID(community.ID())
+	s.Require().NoError(err)
+	s.Require().True(common.IsPubKeyEqual(community.ControlNode(), &s.john.identity.PublicKey))
+
+	var tokenMasterTokenCriteria *protobuf.TokenCriteria
+	for _, permission := range community.TokenPermissions() {
+		if permission.Type == protobuf.CommunityTokenPermission_BECOME_TOKEN_MASTER {
+			s.Require().Len(permission.TokenCriteria, 1)
+			tokenMasterTokenCriteria = permission.TokenCriteria[0]
+			break
+		}
+	}
+	s.Require().NotNil(tokenMasterTokenCriteria)
+
+	s.makeAddressSatisfyTheCriteria(testChainID1, bobAddress, tokenMasterTokenCriteria)
+
+	waitOnAliceCommunityValidation := waitOnCommunitiesEvent(s.alice, func(sub *communities.Subscription) bool {
+		return sub.TokenCommunityValidated != nil
+	})
+	s.advertiseCommunityTo(s.john, community, s.alice)
+	err = <-waitOnAliceCommunityValidation
+	s.Require().NoError(err)
+
+	waitOnBobCommunityValidation := waitOnCommunitiesEvent(s.bob, func(sub *communities.Subscription) bool {
+		return sub.TokenCommunityValidated != nil
+	})
+	s.advertiseCommunityTo(s.john, community, s.bob)
+	err = <-waitOnBobCommunityValidation
+	s.Require().NoError(err)
+
+	s.joinOnRequestCommunity(s.john, community, s.alice)
+	s.joinOnRequestCommunity(s.john, community, s.bob)
+
+	community, err = s.john.communitiesManager.GetByID(community.ID())
+	s.Require().NoError(err)
+	s.Require().True(checkRoleBasedOnThePermissionType(protobuf.CommunityTokenPermission_BECOME_TOKEN_MASTER, &s.bob.identity.PublicKey, community))
+
+	// Simulate control node device changed and new control node device has all members revealed addresses
+	response, err := s.john.PromoteSelfToControlNode(community.ID())
+	s.Require().NoError(err)
+	s.Require().Len(response.CommunityChanges, 1)
+	s.Require().Len(response.CommunityChanges[0].MembersRemoved, 0)
+
+	// Simulate control node device changed and new control node device does not have bob's revealed addresses
+	bobRequestID := communities.CalculateRequestID(s.bob.IdentityPublicKeyString(), community.ID())
+	err = s.john.communitiesManager.RemoveRequestToJoinRevealedAddresses(bobRequestID)
+	s.Require().NoError(err)
+
+	// due to test execution is fast, we update request to join clock
+	clock := uint64(time.Now().Unix() - 2)
```

Comment:
Can we avoid that? I am pretty sure this test will be flaky.. Also,I am struggling to understand why it is needed?